app/src/main/java/net/squanchy/service/firebase/FirebaseAuthService.java

```diff
@@ -86,6 +86,7 @@ private Completable deleteUser(FirebaseUser user) {
 
     public Completable ifUserSignedInThenCompletableFrom(Func1<String, Completable> completableProvider) {
         return ifUserSignedIn()
+                .firstOrError()
```

Comment:
ifUserSignedIn() returns an Observable that doesn't complete, even if it emits a FirebaseUser. At the same time Observable.flatMapCompletable [http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#flatMapCompletable(io.reactivex.functions.Function)](url) pushes data downstream only if the upstream observable completes.

Are you ok if instead I turn ifUserSignedIn() in a Single?